### ISSUE TITLE
Confluence callout not formatted correctly 

### ISSUE THREAD

**Issue by harrylojames** [quarto-dev/quarto-cli#7256]
### Bug description

Publishing a quarto document with a callout results in callouts with a shortened line width. 

Not being able to publish callouts is a major blocker - any guidance would be greatly appreciated!

### Steps to reproduce

Create a file confluence-demo.qmd with the following.

```qmd
---
title: Confluence Demo
format: confluence-html
---

## Overview

::: {.callout-note}
Managing Confluence content with Quarto allows you to author content in Markdown, manage that content with your usual version control tools like Git and GitHub, and leverage Quarto’s tools for including computational output.
:::
```

Publish to a confluence space - 
`quarto publish confluence confluence-demo.qmd
`

### Expected behavior

![image](https://github.com/quarto-dev/quarto-cli/assets/51630653/c05e2d01-4f06-47e3-a949-fa047a17dce8)

### Actual behavior

![image](https://github.com/quarto-dev/quarto-cli/assets/51630653/ba89ed68-2da7-4ada-aee7-092e31e12c4d)

### Your environment

Windows 11 

### Quarto check output

```bash
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.3.450
      Path: C:\Users\hj6\scoop\apps\quarto\current\bin
      CodePage: 1252

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.9.4
      Path: C:/Users/hj6/AppData/Local/Programs/Python/Python39/python.exe
      Jupyter: 5.0.0
      Kernels: 

(\) Checking Jupyter engine render....Exception ignored in: <function BaseEventLoop.__del__ at 0x000001BED3980D30>
Traceback (most recent call last):
  File "C:\Users\hj6\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 683, in __del__
    self.close()
  File "C:\Users\hj6\AppData\Local\Programs\Python\Python39\lib\asyncio\selector_events.py", line 92, in close
    self._close_self_pipe()
  File "C:\Users\hj6\AppData\Local\Programs\Python\Python39\lib\asyncio\selector_events.py", line 99, in _close_self_pipe
    self._remove_reader(self._ssock.fileno())
  File "C:\Users\hj6\AppData\Local\Programs\Python\Python39\lib\asyncio\selector_events.py", line 277, in _remove_reader
    key = self._selector.get_key(fd)
  File "C:\Users\hj6\AppData\Local\Programs\Python\Python39\lib\selectors.py", line 191, in get_key
    return mapping[fileobj]
  File "C:\Users\hj6\AppData\Local\Programs\Python\Python39\lib\selectors.py", line 72, in __getitem__
    fd = self._selector._fileobj_lookup(fileobj)
  File "C:\Users\hj6\AppData\Local\Programs\Python\Python39\lib\selectors.py", line 226, in _fileobj_lookup
    return _fileobj_to_fd(fileobj)
  File "C:\Users\hj6\AppData\Local\Programs\Python\Python39\lib\selectors.py", line 42, in _fileobj_to_fd
    raise ValueError("Invalid file descriptor: {}".format(fd))
ValueError: Invalid file descriptor: -1
[>] Checking Jupyter engine render....OK

[>] Checking R installation...........OK
      Version: 4.1.3
      Path: C:/Users/hj6/R/R-4.1.3
      LibPaths:
        - C:/Users/hj6/R/R-4.1.3/library
      knitr: 1.39
      rmarkdown: 2.14

[>] Checking Knitr engine render......OK
```

**Comment by cderv** [comment]
Thanks for the report. 

This is what I get on my side when I publish to my confluence cloud test instance. 
![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/07e6f154-2dd7-4110-9b67-2d076aacea96)

This is with Quarto 1.4 by the way. This is note the expected callouts. 

Quarto 1.3 reproduce the issue mentionned. 

So something is off in  our Callout. 

@cscheid pinging you as it could be related to crossref.

**Comment by cderv** [comment]
This is the 1.4 xml content 

````xml
<h2 id="overview">Overview<ac:structured-macro ac:name="anchor" ac:schema-version="1" ac:local-id="a6aa6f25-0bee-4a7f-929b-71fcb7eba592" ac:macro-id="d2cb5be1217ae6e086bc60005e9d27b7"><ac:parameter ac:name="">overview</ac:parameter></ac:structured-macro></h2>
<div>
<blockquote>
<p><strong>Note</strong></p>
<p>Managing Confluence content with Quarto allows you to author content in Markdown, manage that content with your usual version control tools like Git and GitHub, and leverage Quarto’s tools for including computational output.</p>
</blockquote>
</div>
````

And this is the expected content (ignoring the multiline issue) 
````xml
<h2 id="overview">Overview<ac:structured-macro ac:name="anchor" ac:schema-version="1" ac:local-id="a6aa6f25-0bee-4a7f-929b-71fcb7eba592" ac:macro-id="d2cb5be1217ae6e086bc60005e9d27b7"><ac:parameter ac:name="">overview</ac:parameter></ac:structured-macro></h2>
<ac:structured-macro ac:name="info" ac:schema-version="1" ac:macro-id="1c8062cd-87de-4701-a698-fd435e057468"><ac:rich-text-body><p>Managing Confluence content with Quarto allows you to author content
in Markdown, manage that content with your usual version control tools
like Git and GitHub, and leverage Quarto’s tools for including
computational output.</p></ac:rich-text-body></ac:structured-macro>
````

Callouts are not process as they should...

**Comment by cscheid** [comment]
There's something clearly wrong, but I don't know why our test suite isn't catching it.

**Comment by cderv** [comment]
Now that the confluence callout regression from 1.3 to 1.4 is fixed, we can look at the initial issue. 

I'll reopen to look into the wrapping to new line.

**Comment by cderv** [comment]
@harrylojames the line width issue should now be solved. This will be available in quarto 1.4.

**Comment by harrylojames** [comment]
@cderv Thank you very much all for the rapid turn around!
